### ISSUE TITLE
Encoding.stringFromFields in 0.14.0 throws Error (bytesFromFields): Invalid encoding

### ISSUE THREAD

**Issue by dfstio** [o1-labs/o1js#1233]
The Encoding.stringFromFields that used to work as expected in 0.13.0 throws an error in 0.14.0:
```
 FAIL  experimental/encoding_error.test.ts
  Convert string to Fields and back
    ✓ should convert string to Fields (1 ms)
    ✕ should convert Fields to string

  ● Convert string to Fields and back › should convert Fields to string

    Error (bytesFromFields): Invalid encoding.

      12 |   it("should convert Fields to string", async () => {
      13 |     expect(fields.length).toBe(1);
    > 14 |     const restoredString: string = Encoding.stringFromFields(fields);
         |                                             ^
      15 |     expect(restoredString).toBe(str);
      16 |   });
      17 | });

      at bytesFromFields (.yarn/cache/o1js-npm-0.14.0-990b550bc5-7f46d3b4c8.zip/node_modules/o1js/dist/node/index.cjs:11885:11)
      at Object.stringFromFields (.yarn/cache/o1js-npm-0.14.0-990b550bc5-7f46d3b4c8.zip/node_modules/o1js/dist/node/index.cjs:11857:15)
      at Object.<anonymous> (experimental/encoding_error.test.ts:14:45)
```

Test code:
```typescript
import { describe, expect, it } from "@jest/globals";
import { Encoding, Field } from "o1js";

const str = "Hello, World!";
let fields: Field[] = [];

describe("Convert string to Fields and back", () => {
  it("should convert string to Fields", async () => {
    fields = Encoding.stringToFields(str);
    expect(fields.length).toBe(1);
  });
  it("should convert Fields to string", async () => {
    expect(fields.length).toBe(1);
    const restoredString: string = Encoding.stringFromFields(fields);
    expect(restoredString).toBe(str);
  });
});

```

**Comment by dfstio** [comment]
Temporarily solution uses chunk length encoded into byte instead of STOP byte:
```typescript
import { describe, expect, it } from "@jest/globals";
import { Field } from "o1js";

const str = `Although it is possible to set the verification key inside the SmartContract’s method by passing the variable of the type VerificationKey to the method, it is not possible to set zkAppUri inside the SmartContract’s method by passing the variable of the type Types.ZkappUri to the method.`;
let fields: Field[] = [];

describe("Convert string to Fields and back", () => {
  it("should convert string to Fields", async () => {
    fields = stringToFields(str);
    console.log(fields.length);
    fields.forEach((field) => {
      expect(field.toBigInt()).toBeLessThanOrEqual(BigInt(31) << BigInt(240));
    });
  });

  it("should convert Fields to string", async () => {
    const str1 = stringFromFields(fields);
    expect(str1).toEqual(str);
  });
});

function bigIntToUint8Array(value: bigint, length: number): Uint8Array {
  const bytes = new Uint8Array(length);
  for (let i = length - 1; i >= 0; i--) {
    bytes[i] = Number(value & BigInt(0xff));
    value >>= BigInt(8);
  }
  return bytes;
}

// Helper function to convert Uint8Array to BigInt
function uint8ArrayToBigInt(bytes: Uint8Array): bigint {
  let value = BigInt(0);
  bytes.forEach((byte) => {
    value = (value << BigInt(8)) + BigInt(byte);
  });
  return value;
}

// convert string to chunks of 30 chars
function stringToFields(str: string): Field[] {
  const encoder = new TextEncoder();
  const bytes = encoder.encode(str);
  const fields: Field[] = [];

  for (let i = 0; i < bytes.length; i += 30) {
    const chunkSize = Math.min(30, bytes.length - i);
    const chunk = new Uint8Array(31);
    chunk[0] = chunkSize; // The first byte is the length of the actual string chunk
    chunk.set(bytes.slice(i, i + chunkSize), 31 - chunkSize);
    fields.push(Field.from(uint8ArrayToBigInt(chunk)));
  }

  return fields;
}

// convert Field[] to string
function stringFromFields(fields: Field[]): string {
  const decoder = new TextDecoder();
  let str = "";

  fields.forEach((field) => {
    const bigIntValue = field.toBigInt();
    // The first byte is the length, so shift right by 240 bits (30 bytes) to get it
    const length = Number(bigIntValue >> BigInt(240));
    const chunk = bigIntToUint8Array(bigIntValue, length + 1).slice(1); // Remove the length byte
    const str1 = decoder.decode(chunk);
    str += str1;
  });

  return str;
}

```

**Comment by dfstio** [comment]
Fixed in 0.14.1
